package/libnetwork/src/libnetwork.c

```diff
@@ -377,20 +447,25 @@ void ntrip_download(const network_config_t *config)
     .debug = config->debug,
   };
 
+  CURL *curl = network_setup();
+  if (curl == NULL) {
+    return;
+  }
+
   network_progress_t progress = {
     .bytes = 0,
     .count = 0,
     .debug = config->debug,
+    .curl = curl,
   };
 
-  CURL *curl = network_setup();
-  if (curl == NULL) {
-    return;
-  }
-
   struct curl_slist *chunk = ntrip_init(curl);
 
+  curl_easy_setopt(curl, CURLOPT_PUT,              1L);
+  curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST,    "GET");
   curl_easy_setopt(curl, CURLOPT_URL,              config->url);
+  curl_easy_setopt(curl, CURLOPT_READFUNCTION,     network_upload_write);
```

Comment:
Soo.... this always runs?

Reply:
yeah, we were thinking that as long as we don't find adverse behavior with NTRIP clients that don't need this sent up, we should just leave it on always and make ntrip operation a little less painful for the user. Talked to lance and he also doesn't think this is a problem to leave always on. Fairly easy change to make later if this proves incorrect.

Reply:
I bet its problematic. Consider an ntrip server that never reads its pipe and only writes - that should not be healthy for the tcp connection. Would rather see this be incremental and only impact those using the feature.